### ISSUE TITLE
Ubuntu 24.04 Noble Numbat

### ISSUE THREAD

**Issue by ojura** [ubi-agni/ros-builder-action#7]
Is there a chance 24.04 gets supported at https://ros.packages.techfak.net/?

**Comment by rhaschke** [comment]
I triggered a build: https://github.com/ubi-agni/ros-builder-action/actions/runs/8848336789/job/24297968977
Let's see how far it gets... I won't have many resources to fix issues.

**Comment by rhaschke** [comment]
Many projects finished successfully. They are available on ros.packages.techfak.net now.

**Comment by ojura** [comment]
Thanks!

rosbag is a fairly important one. As far as I can see, it just needs a define to build, `BOOST_TIMER_ENABLE_DEPRECATED`. Should I open a MR at https://github.com/ubi-agni/ros_comm.git?

**Comment by rhaschke** [comment]
Yes, please open a MR against the obese-devel branch

**Comment by ojura** [comment]
Thanks for merging it. As far as I can see, the latest pipeline run didn't properly get to the actual build stage for rosbag and now failed because it couldn't find catkin?

https://github.com/ubi-agni/ros-builder-action/actions/runs/8975224192/job/24676002410#step:5:2093

**Comment by rhaschke** [comment]
Yeah, I'm not yet sure, why existing packages are not found by rosdep...

**Comment by ojura** [comment]
Any way to move forward? Maybe just flush all packages and try again?

**Comment by rhaschke** [comment]
I disabled the rosdep check and then it worked?!? `rosbag` should be available for Noble now as well.

**Comment by ojura** [comment]
Hi @rhaschke, would you mind triggering another run for Noble? I’m trying to see what happened for sensor_msgs, but the logs from May were deleted. Thanks!

**Comment by rhaschke** [comment]
Here you go: https://github.com/ubi-agni/ros-builder-action/actions/runs/10506664555

**Comment by rhaschke** [comment]
I imported the newly-built packages into the apt repo now. sensor-msgs should be available.

**Comment by ojura** [comment]
Thanks @rhaschke!

Btw, octomap is one dependency causing a few failures in the build tree, because of an iterator deprecation warning treated as an error. 
```
   /<<BUILDDIR>>package/include/octomap/OcTreeIterator.hxx:42:39: error: ‘template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator’ is deprecated [-Werror=deprecated-declarations]
     42 |     class iterator_base : public std::iterator<std::forward_iterator_tag, NodeType>{
        |                                       ^~~~~~~~
  In file included from /usr/include/c++/13/bits/stl_iterator_base_funcs.h:66,
                   from /usr/include/c++/13/string:47,
                   from /usr/include/c++/13/bits/locale_classes.h:40,
                   from /usr/include/c++/13/bits/ios_base.h:41,
                   from /usr/include/c++/13/ios:44,
                   from /usr/include/c++/13/istream:40,
                   from /usr/include/c++/13/fstream:40,
                   from /<<BUILDDIR>>package/include/octomap/AbstractOcTree.h:38,
                   from /<<BUILDDIR>>package/src/AbstractOcTree.cpp:35:
  /usr/include/c++/13/bits/stl_iterator_base_types.h:127:34: note: declared here
    127 |     struct _GLIBCXX17_DEPRECATED iterator
        |                                  ^~~~~~~~
```
We are building 1.9.8-1 from https://github.com/ros-gbp/octomap-release/ obtained by bloom-release.

However, upstream released 1.9.10 in March, which probably updated this.
https://github.com/OctoMap/octomap/tags

Do you think we could bump this to the new upstream?
https://github.com/ubi-agni/ros-builder-action/blob/09830187e19b6ea9d22e1828fb8a7dc706d33238/ros-one.repos#L342-L345

I'm not sure if it's a trivial effort, or one needs to fork and perform bloom-release for 1.9.10.

**Comment by rhaschke** [comment]
> However, upstream released 1.9.10 in March.

Version 1.9.10 does not exist. There is only 1.10.0, which is not compatible to versions 1.9.x.
As fcl uses the system-installed octomap, which is 1.9.7, we cannot switch to the incompatible version 1.10.0.
The way to go, is to use the system-installed version everywhere (see https://github.com/ros/rosdistro/issues/41622).
